### ISSUE TITLE
Bun workspaces doesn't work

### ISSUE THREAD

**Issue by robobun** [oven-sh/bun#5181]
Hi, according to [this](https://bun.sh/guides/install/workspaces) guide, when you run ``bun install`` in project root with workspaces it installs all packages from all workspace. 
But for me it only installs 2 workspace packages.
Also ``bun install`` in one of workspaces doesn't work, it says 
```
error: package "stuff-a" not found registry.npmjs.org/stuff-a 404
error: stuff-a@* failed to resolve
```
Also, it creates seperate bun.lockb and node_modules but it shouldn't with all other external dependencies but it shouldn't.

_Originally reported on Discord: [`Bun workspaces doesn't work`](https://discord.com/channels/876711213126520882/1151165617714511882/1151165617714511882)_

**Comment by eksno** [comment]
Hey! Could you try importing `"stuff-a": "workspace:*"` as specified in [The Bun Docs](https://bun.sh/docs/install/workspaces) instead of `"stuff-a": "*"` and see if that works for you?

I tested it myself and following the instructions in the guide you sent didn't work for me either, it seems the docs you referred to are outdated.

Assuming the fix works for you I recommend using [these docs](https://bun.sh/docs) moving forward.

**Comment by vsviridov** [comment]
I've tried it with `workspace:*` syntax and while `bun install` in root works, any package operations inside the package folder fail citing `unknown package` and a 404 error from npm.

**Comment by mpotane** [comment]
> I've tried it with `workspace:*` syntax and while `bun install` in root works, any package operations inside the package folder fail citing `unknown package` and a 404 error from npm.

same it can't recognize workspace path correctly

**Comment by artecoop** [comment]
I've stumbled on a related problem: if I rename one of the packages, bun doesn't find it anymore, throwing the same error OP described. After nuking, bun install finds all the packages. This is rather annoying

**Comment by Merded** [comment]
> Hey! Could you try importing `"stuff-a": "workspace:*"` as specified in [The Bun Docs](https://bun.sh/docs/install/workspaces) instead of `"stuff-a": "*"` and see if that works for you?
> 
> I tested it myself and following the instructions in the guide you sent didn't work for me either, it seems the docs you referred to are outdated.
> 
> Assuming the fix works for you I recommend using [these docs](https://bun.sh/docs) moving forward.

It only works when I do something like ``workspace:../packages/stuff-a``.

**Comment by dylan-conway** [comment]
The bugs in this issue were fixed in bun `v1.1.8` and `v1.1.9`. Reopen or create a new issue if you continue to have these bugs with the latest release of bun
